examples/v1beta1/al2-kubelet-log-query.yaml

```diff
@@ -0,0 +1,49 @@
+# This example NodePool will provision instances using the AL2 EKS-Optimized AMI
+# and will be prepended to a Karpenter managed section that will bootstrap the kubelet.
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: default
+spec:
+  template:
+    spec:
+      nodeClassRef:
+        name: al2
+---
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+metadata:
+  name: al2
+spec:
+  amiFamily: AL2
+  role: "KarpenterNodeRole-${CLUSTER_NAME}" # replace with your cluster name
+  subnetSelectorTerms:
+    - tags:
+        karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  securityGroupSelectorTerms:
+    - tags:
+        karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  userData: |
+    MIME-Version: 1.0
+    Content-Type: multipart/mixed; boundary="BOUNDARY"
+
+    --BOUNDARY
+    Content-Type: text/x-shellscript; charset="us-ascii"
+
+    #!/bin/bash
+
+    set -e
+
+    # Add additional KUBELET_EXTRA_ARGS to the service
+    # Requires Kubernetes 1.27 (alpha feature)
+    cat << EOF > /etc/systemd/system/kubelet.service.d/90-kubelet-extra-args.conf
+    [Service]
+    Environment="KUBELET_EXTRA_ARGS=--feature-gates=NodeLogQuery=true $KUBELET_EXTRA_ARGS"
+    EOF
+    systemctl daemon-reload
+
+    # Enable log handler and log query to the kubelet configuration
+    echo "$(jq '.enableSystemLogHandler=true' /etc/kubernetes/kubelet/kubelet-config.json)" > /etc/kubernetes/kubelet/kubelet-config.json
+    echo "$(jq '.enableSystemLogQuery=true' /etc/kubernetes/kubelet/kubelet-config.json)" > /etc/kubernetes/kubelet/kubelet-config.json
+
```

Comment:
Any need to restart the kubelet here / can we always assume that user data runs first?